chromadb/utils/embedding_functions/onnx_mini_lm_l6_v2.py

```diff
@@ -211,26 +212,29 @@ def _download_model_if_not_exists(self) -> None:
             "vocab.txt",
         ]
         extracted_folder = os.path.join(self.DOWNLOAD_PATH, self.EXTRACTED_FOLDER_NAME)
-        onnx_files_exist = True
-        for f in onnx_files:
-            if not os.path.exists(os.path.join(extracted_folder, f)):
-                onnx_files_exist = False
-                break
-        # Model is not downloaded yet
-        if not onnx_files_exist:
-            os.makedirs(self.DOWNLOAD_PATH, exist_ok=True)
-            if not os.path.exists(
-                os.path.join(self.DOWNLOAD_PATH, self.ARCHIVE_FILENAME)
-            ) or not _verify_sha256(
-                os.path.join(self.DOWNLOAD_PATH, self.ARCHIVE_FILENAME),
-                self._MODEL_SHA256,
-            ):
-                self._download(
-                    url=self.MODEL_DOWNLOAD_URL,
-                    fname=os.path.join(self.DOWNLOAD_PATH, self.ARCHIVE_FILENAME),
-                )
-            with tarfile.open(
-                name=os.path.join(self.DOWNLOAD_PATH, self.ARCHIVE_FILENAME),
-                mode="r:gz",
-            ) as tar:
-                tar.extractall(path=self.DOWNLOAD_PATH)
+        lock_file = os.path.join(self.DOWNLOAD_PATH, ".lock")
+
+        with portalocker.Lock(lock_file):
```

Comment:
Can this be accomplished without portalocked / what is this addressing (so I can help think of how else we can do it?)

Reply:
still in draft, just wanted to see if this would fix a race condition haha

Reply:
if there are multiple instances of Chroma active on the same system (which is a valid usecase as long as they have different persistence directories) there's a possible race condition where >1 instance are downloading the model at the same time

We could make this only import & use the lock when running in CI. I would slightly prefer to always enable it, as it does fix a possible bug (although the likelihood of it happening outside of tests seems fairly low). 

There's enough platform-specific logic around file locking that I would prefer not to handle it ourselves.

https://github.com/tox-dev/filelock is another option that seems to be a bit more active